23 user customization

NotJustJoe opened the pull request.

NotJustJoe committed:
Merge remote-tracking branch 'origin/42-register-email' into 23-user-customization

NotJustJoe committed:
Merge pull from 42-register-email: resolve conflict in shared/index.ts

NotJustJoe committed:
feat: implemented user customization

NotJustJoe committed:
refactor: simplify validation functions and improve error handling in registration form

NotJustJoe committed:
Merge branch '42-register-email' into 23-user-customization

NotJustJoe committed:
Add admin middleware, user role to model, and migration for role column; update i18n initialization

NotJustJoe committed:
Add admin role handling and create admin panel route; update localization for admin dashboard

NotJustJoe committed:
Merge remote-tracking branch 'origin' into 23-user-customization

NotJustJoe committed:
WIP: enhance user profile functionality and comments section

- Added UserComments component to display user comments on profile pages.
- Implemented API functions to fetch user comments and user profiles by username.
- Updated account settings page to include profile picture upload functionality.
- Refactored Navbar component to dynamically display user profile links.
- Improved localization files to support new profile-related strings.
- Created new user profile page to display user information and activity.
- Removed admin dashboard route as it is not currently implemented.
- Updated User model to replace role with isAdmin boolean for better clarity.
- Introduced UserMovieAction enum for better handling of user actions on movies.

NotJustJoe committed:
refactor: clean up whitespace in various files for consistency

Thomas0642 committed:
feat: implement profile picture processing and user update validation

Thomas0642 committed:
feat: enhance localization for comments and profile sections in Arabic, Spanish, French, and Chinese, fixed missing translation keys

Thomas0642 committed:
feat: implement profile picture upload and processing, enhance user profile update functionality

NotJustJoe committed:
feat: add user profile picture component and enhance profile picture upload functionality

NotJustJoe committed:
feat: enhance user avatar and profile picture components with clickable usernames and improved display logic

NotJustJoe committed:
refactor: clean up UserAvatar component code for improved readability and consistency

LaBoiteDeCarton commented:
<img width="1258" height="478" alt="Capture d’écran 2025-09-25 à 10 38 07" src="https://github.com/user-attachments/assets/1c0ac433-1404-448f-ab94-f77b1fc88f25" />
La migration est peté.


J'ai aussi test de wipe le volume de la db et j'ai relancé la migration.

NotJustJoe committed:
refactor: optimize migration logic for adding and dropping OAuth fields in users table

LaBoiteDeCarton commented on apps/front/src/lib/api/users.ts:
```diff
@@ -0,0 +1,56 @@
+) {
+	if (!browser) return;
+
+	const response = await fetch(`http://localhost:3333/users/${userId}`, {
```
faut pas mettre localhost mais la variable d'env

LaBoiteDeCarton commented on apps/front/src/lib/services/api.ts:
```diff
@@ -378,4 +404,69 @@ export async function updateComment(
+		console.error('Error updating user profile:', error);
+		throw error;
+	}
+}
```
Cette partie n'est pas utilisé. A supprimer.
Ici on appel un endpoint qui est censé etre appelé par le consommateur de l'api et non le front (faudrait preparer une refont pour créer un middleware afin de proteger les routes a un utilisateur RestfulApi je crois)

LaBoiteDeCarton commented on apps/front/src/lib/locales/ar.json:
```diff
@@ -23,7 +23,13 @@
 		"save": "حفظ",
```
La page de l'account n'ets pas traduit entierement

<img width="316" height="47" alt="Capture d’écran 2025-09-25 à 15 54 57" src="https://github.com/user-attachments/assets/603a5ac0-cbc3-4634-bbaa-66942062904a" />
<img width="563" height="667" alt="Capture d’écran 2025-09-25 à 15 55 04" src="https://github.com/user-attachments/assets/041b46e2-3188-45ee-be73-9e5d4f366795" />
<img width="291" height="65" alt="Capture d’écran 2025-09-25 à 15 55 09" src="https://github.com/user-attachments/assets/6a8b1f5f-090e-43f5-b5c0-b0d29adf93ef" />

LaBoiteDeCarton commented on apps/front/src/lib/components/tadflix/comments/UserAvatar.svelte:
```diff
@@ -3,50 +3,59 @@
 		username: string;
 		size?: 'small' | 'medium' | 'large';
 		class?: string;
+		clickable?: boolean;
+		onclick?: () => void;
 	}
 
-	const { username, size = 'medium', class: className = '' }: UserAvatarProps = $props();
+	const {
+		username,
+		size = 'medium',
+		class: className = '',
+		clickable = false,
+		onclick
+	}: UserAvatarProps = $props();
 
 	// Generate avatar based on username initials
 	const getInitials = (name: string): string => {
 		return name
```
Le logo choisi ne s'affiche pas sur les commentaire 

<img width="196" height="116" alt="Capture d’écran 2025-09-25 à 16 00 18" src="https://github.com/user-attachments/assets/123485f4-e762-409e-8bc1-9354f9125e91" />

LaBoiteDeCarton reviewed (changes requested):
L'ensemble est fonctionnel. Quelques details a corriger et voir ce que l'on fait pour le 'restful' (autre ticket etc)

NotJustJoe committed:
feat(admin): add admin dashboard for managing downloaded movies

- Implemented routes for admin access to manage movies.
- Added a new admin page with functionality to list, delete, and manage downloaded movies.
- Integrated authentication checks to ensure only admin users can access the admin features.
- Updated localization files to include admin-related strings in multiple languages.
- Enhanced the navbar to conditionally display the admin link for admin users.

Thomas0642 committed:
feat: add profile picture support to comments and user components

Thomas0642 committed:
feat: add internationalization support for comments section with new translations

Thomas0642 committed:
refactor: remove unused updateUserProfile function from users API

Thomas0642 committed:
feat: enhance profile picture upload with error handling and internationalization support

Commit:
Merge branch '23-user-customization' of github.com:asebrech/Hypertube into 23-user-customization

Thomas0642 committed:
feat: add back navigation button to user profile with improved layout

Commit:
update movielist of viewed list in profile

Commit:
Merge branch '23-user-customization' of github.com:asebrech/Hypertube into 23-user-customization

LaBoiteDeCarton committed:
Merge branch 'staging' into 23-user-customization

Commit:
max height for comment section update to none

Commit:
Merge branch '23-user-customization' of github.com:asebrech/Hypertube into 23-user-customization

Commit:
add data for token

LaBoiteDeCarton reviewed (approved):
good
